components/support/ffi/src/lib.rs

```diff
@@ -343,3 +343,81 @@ fn init_backtraces_once() {
 
 #[cfg(not(feature = "log_backtraces"))]
 fn init_backtraces_once() {}
+
+/// ByteBuffer is a struct that represents an array of bytes to be sent over the FFI boundaries.
+///
+/// ## Caveats
+///
+/// Note that the order of the fields is `len` (an i64) then `data` (a `*mut u8`), getting
+/// this wrong on the other side of the FFI will cause memory corruption and crashes.
+///
```

Comment:
Also, a note that it should not be used to pass data into rust unless the ByteBuffer was allocated by rust code (because ByteBuffer assumes it owns the underlying data, and will Drop it).